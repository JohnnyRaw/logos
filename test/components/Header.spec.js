import React from 'react';
import { mount } from 'enzyme';

import Header from 'components/Header';
import { appState } from 'reducers/app';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    app: appState,
    dispatch: mockDispatch,
    location: {
      pathname: '/'
    }
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    wrapper.find('.app__header__logo').simulate('click');
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: '@@router/CALL_HISTORY_METHOD',
      payload: { method: 'push', args: [{ pathname: '/', search: undefined, state: undefined }] }
    });
  });
});

