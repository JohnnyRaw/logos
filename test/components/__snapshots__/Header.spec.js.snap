exports[`Header should render properly 1`] = `"<header class="app__header"><div class="app__container"><a href="/" class="app__header__logo"><div class="app__logo"><span class="isvg failed app__logo__large"></span><span class="isvg failed app__logo__icon"></span></div></a><div class="app__header__search"><input type="text" name="search" value=""><span class="input-icon"><i class="i-search"></i></span></div><div class="app__header__social"><a href="https://twitter.com/intent/follow?screen_name=svgporn" target="_blank"><i class="i-twitter"></i></a><a href="https://www.facebook.com/svgporn" target="_blank"><i class="i-facebook"></i></a></div></div></header>"`;
